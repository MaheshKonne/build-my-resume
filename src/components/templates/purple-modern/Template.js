import React, {Fragment} from "react";
import styled from "styled-components";
import data from '../../../static/data';
import Header from "../../Header";
import {theme} from "../../../lib/theme";
import Diamond from "../../SquareDiamond";
import Content from "../../Content";
import Chip from "@material-ui/core/Chip";
import {FlexCell, FlexContainer} from "../../FlexBox";

const TemplateHeader = styled.header`
    padding: 40px;
    background: ${props => props.theme.primary};
    border-radius: 5px 5px 0 0;
    text-align: center;
    color: ${theme.white};
`;

const TemplateFooter = styled.footer`
    height: 40px;
    background: ${props => props.theme.primary};
    border-radius: 0 0 5px 5px;
    color: white;
`;

const FooterWrapper = styled.div`
    height: 100%;
    display: flex;
    justify-content: space-around;
    align-items: center;
`;

const TemplateBody = styled.section`
    border: 2px solid ${props => props.theme.quaternary};
`;

const Section = styled.section`
    background-color: ${props => props.background}
`;

const ContentWrapper = styled.section`
    padding: 30px 50px;
`;

const FlexContainerElem = FlexContainer({
    "padding-top": "10px",
    "margin-bottom": "30px"
})('div');

const FlexCellElem = FlexCell({
    "min-width": "300px",
    "margin-right": "30px"
})('div');

const Template = ({template='canva-purple-modern-resume'}) => {
    const selectedTheme = theme[template];

    return (
        <Fragment>
            <TemplateHeader theme={selectedTheme}>
                <Header color={theme.white}>{data.name}</Header>
                <Header color={theme.white} heading="h4">{data.role}</Header>
            </TemplateHeader>
            <TemplateBody theme={selectedTheme}>
                <ContentWrapper>
                    <Header heading="h2" color={selectedTheme.primary} letterSpacing="3px">{data.profile.title}</Header>
                    <Content>{data.profile.description}</Content>
                </ContentWrapper>
                <Section background={selectedTheme.quaternary}>
                    <ContentWrapper>
                        <Header heading="h2" color={selectedTheme.primary} letterSpacing="3px">{data.experience.title}</Header>
                        {data.experience.details.map((item, i) => (
                            <FlexContainerElem key={`${item.company}-${i}`}>
                                <FlexCellElem>
                                    <Content>{item.role}</Content>
                                    <Content>{item.company}</Content>
                                    <Content>{item.duration}</Content>
                                </FlexCellElem>
                                <Content>{item.description}</Content>
                            </FlexContainerElem>
                        ))}
                    </ContentWrapper>
                </Section>
                <ContentWrapper>
                    <Header heading="h2" color={selectedTheme.primary} letterSpacing="3px">{data.education.title}</Header>
                    {data.education.details.map(item => (
                        <FlexCellElem key={item.year}>
                            <Content>{item.course}</Content>
                            <Content>{item.university}</Content>
                            <Content>{item.year}</Content>
                            <Content>{item.grade}</Content>
                        </FlexCellElem>
                    ))}
                </ContentWrapper>
                <ContentWrapper>
                    <Header heading="h2" color={selectedTheme.primary} letterSpacing="3px">SKILLS</Header>
                    {data.skills.map(skill => <Chip key={skill} variant="outlined" color="red" label={skill} />)}
                </ContentWrapper>
            </TemplateBody>
            <TemplateFooter theme={selectedTheme}>
                <FooterWrapper>
                    <Header heading="h4" color={theme.white}>{data.contact.email}</Header>
                    <Diamond />
                    <Header heading="h4" color={theme.white}>{data.contact.website}</Header>
                    <Diamond />
                    <Header heading="h4" color={theme.white}>{data.contact.phone}</Header>
                </FooterWrapper>
            </TemplateFooter>
        </Fragment>
    );
};

export default Template;