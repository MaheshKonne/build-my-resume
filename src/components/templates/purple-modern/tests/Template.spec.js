import React from "react";
import renderer from 'react-test-renderer';
import Template from "../Template";



describe('<Template />', () => {
    it('should match with snapshot', () => {
        const component = renderer.create(<Template />).toJSON();
        expect(component).toMatchSnapshot();
    });
});