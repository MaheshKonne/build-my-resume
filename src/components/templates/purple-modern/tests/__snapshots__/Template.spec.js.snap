// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Template /> should match with snapshot 1`] = `
Array [
  .c0 {
  padding: 40px;
  background: #7c4373;
  border-radius: 5px 5px 0 0;
  text-align: center;
  color: #fff;
}

.c1 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #fff;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c2 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #fff;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<header
    className="c0"
  >
    <h1
      className="c1"
    >
      Mahesh Konne
    </h1>
    <h4
      className="c2"
    >
      UI Developer
    </h4>
  </header>,
  .c3 {
  margin: 0;
  color: #000;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  border: 2px solid #eee9ee;
}

.c4 {
  background-color: #eee9ee;
}

.c1 {
  padding: 30px 50px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 10px;
  margin-bottom: 30px;
}

.c7 {
  min-width: 300px;
  margin-right: 30px;
}

.c2 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #7c4373;
  -webkit-letter-spacing: 3px;
  -moz-letter-spacing: 3px;
  -ms-letter-spacing: 3px;
  letter-spacing: 3px;
}

.c5 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #7c4373;
  -webkit-letter-spacing: 3px;
  -moz-letter-spacing: 3px;
  -ms-letter-spacing: 3px;
  letter-spacing: 3px;
}

.c8 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #7c4373;
  -webkit-letter-spacing: 3px;
  -moz-letter-spacing: 3px;
  -ms-letter-spacing: 3px;
  letter-spacing: 3px;
}

.c9 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #7c4373;
  -webkit-letter-spacing: 3px;
  -moz-letter-spacing: 3px;
  -ms-letter-spacing: 3px;
  letter-spacing: 3px;
}

<section
    className="c0"
  >
    <section
      className="c1"
    >
      <h2
        className="c2"
      >
        PROFILE
      </h2>
      <p
        className="c3"
        color="#000"
        letterSpacing="normal"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
      </p>
    </section>
    <section
      className="c4"
    >
      <section
        className="c1"
      >
        <h2
          className="c5"
        >
          PROFESSIONAL EXPERIENCE
        </h2>
        <div
          className="c6"
        >
          <div
            className="c7"
          >
            <p
              className="c3"
              color="#000"
              letterSpacing="normal"
            >
              UI Developer
            </p>
            <p
              className="c3"
              color="#000"
              letterSpacing="normal"
            >
              ThoughtWorks
            </p>
            <p
              className="c3"
              color="#000"
              letterSpacing="normal"
            >
              2018 - Present
            </p>
          </div>
          <p
            className="c3"
            color="#000"
            letterSpacing="normal"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
          </p>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
          >
            <p
              className="c3"
              color="#000"
              letterSpacing="normal"
            >
              MEAN Stack Developer
            </p>
            <p
              className="c3"
              color="#000"
              letterSpacing="normal"
            >
              UNO Career
            </p>
            <p
              className="c3"
              color="#000"
              letterSpacing="normal"
            >
              2017 - 2018
            </p>
          </div>
          <p
            className="c3"
            color="#000"
            letterSpacing="normal"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
          </p>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
          >
            <p
              className="c3"
              color="#000"
              letterSpacing="normal"
            >
              UX Engineer
            </p>
            <p
              className="c3"
              color="#000"
              letterSpacing="normal"
            >
              Divami Design Labs
            </p>
            <p
              className="c3"
              color="#000"
              letterSpacing="normal"
            >
              2015 - 2017
            </p>
          </div>
          <p
            className="c3"
            color="#000"
            letterSpacing="normal"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
          </p>
        </div>
      </section>
    </section>
    <section
      className="c1"
    >
      <h2
        className="c8"
      >
        EDUCATION
      </h2>
      <div
        className="c7"
      >
        <p
          className="c3"
          color="#000"
          letterSpacing="normal"
        >
          Computer Science Engineering
        </p>
        <p
          className="c3"
          color="#000"
          letterSpacing="normal"
        >
          IIIT Basar
        </p>
        <p
          className="c3"
          color="#000"
          letterSpacing="normal"
        />
        <p
          className="c3"
          color="#000"
          letterSpacing="normal"
        >
          CGPA: 8.2
        </p>
      </div>
    </section>
    <section
      className="c1"
    >
      <h2
        className="c9"
      >
        SKILLS
      </h2>
      <div
        className="MuiChip-root MuiChip-outlined"
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label"
        >
          Javascript
        </span>
      </div>
      <div
        className="MuiChip-root MuiChip-outlined"
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label"
        >
          React
        </span>
      </div>
      <div
        className="MuiChip-root MuiChip-outlined"
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label"
        >
          HTML
        </span>
      </div>
      <div
        className="MuiChip-root MuiChip-outlined"
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label"
        >
          CSS
        </span>
      </div>
    </section>
  </section>,
  .c3 {
  width: 10px;
  height: 10px;
  background: #fff;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c0 {
  height: 40px;
  background: #7c4373;
  border-radius: 0 0 5px 5px;
  color: white;
}

.c1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #fff;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c4 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #fff;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c5 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #fff;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<footer
    className="c0"
  >
    <div
      className="c1"
    >
      <h4
        className="c2"
      >
        maheshkonneiiitb@gmail.com
      </h4>
      <div
        className="c3"
        color="#fff"
        size={10}
      />
      <h4
        className="c4"
      >
        http://maheshkonne.me
      </h4>
      <div
        className="c3"
        color="#fff"
        size={10}
      />
      <h4
        className="c5"
      >
        +91 9999999999
      </h4>
    </div>
  </footer>,
]
`;
